presto-orc/src/main/java/com/facebook/presto/orc/reader/ListFilter.java

```diff
@@ -379,7 +379,7 @@ private static long[] extractFilters(Map<Subfield, TupleDomainFilter> filters, i
         long[] filterCodes = new long[maxSubscript + 1];
         for (int i = 0; i < subscripts.length; i++) {
             if (subscripts[i] != -1) {
-                filterCodes[subscripts[i]] |= 1 << i;
+                filterCodes[subscripts[i]] |= 1L << i;
```

Comment:
Does this mean it will now overflow at 63 instead of 31 ? Do we have any metric on how many queries are actually hitting this limit ?

Reply:
> Does this mean it will now overflow at 63 instead of 31 ?

Yes. But ListFilter only allows up to 64 entries, it's not going to be hit.

Here is what it has in the constructor:
```
checkArgument(subfieldFilters.size() <= 64, "Number of filters cannot exceed 64");
subscriptFilters = extractFilters(subfieldFilters, level, subfield -> subfield.getPath().size() > level);
```

> Do we have any metric on how many queries are actually hitting this limit ?

Sorry, not my area of expertise. But it's likely very-very small number because filtering list entries is not particularly popular operation.